### PR TITLE
develop: optimize overview and spring boot

### PR DESCRIPTION
**Pull request by ran-huang**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-sample-application-spring-boot.md [hunk 1]
```diff
@@ -567,28 +511,28 @@ curl --location --request PUT 'http://localhost:8080/player/trade' \
   --data-urlencode 'price=100'
 ```
 
-We use **Form Data** as the payload of our message with the request parameters are sell player ID `sellID`, buy player ID `buyID`, number of goods purchased `amount`, number of coins consumed for purchase `price`.
+The request uses **Form Data** as the payload. The example request indicated that the seller's ID (`sellID`) is 1, the buyer's ID (`buyID`) is 2, the number of goods purchased (`amount`) is 10, and the number of coins consumed for purchase (`price`) is 100.
```
Comment (hfxsd) [inline develop/dev-guide-sample-application-spring-boot.md:514]:
```suggestion
The request uses **Form Data** as the payload. The example request indicates that the seller's ID (`sellID`) is 1, the buyer's ID (`buyID`) is 2, the number of goods purchased (`amount`) is 10, and the number of coins consumed for purchase (`price`) is 100.
```

File: develop/dev-guide-sample-application-spring-boot.md [hunk 2]
```diff
@@ -567,28 +511,28 @@ curl --location --request PUT 'http://localhost:8080/player/trade' \
...
 
-The return value is whether the transaction is successful or not. When there is insufficient goods for the selling player, insufficient gold for the buying player or database error, the transaction will not be successful and no player's gold or goods will be lost due to the database [transaction](/develop/dev-guide-transaction-overview.md) guarantee.
+The return value is whether the transaction is successful or not. When there are insufficient goods for the seller, insufficient coins for the buyer, or a database error, the [database transaction](/develop/dev-guide-transaction-overview.md) guarantees that the trade is not successful and no player's coins or goods are lost.
 
 ```json
 true
 ```
 
-### Step 6.3 Request with Shell Script
+### Step 6.3 Request with Shell script
```
Comment (hfxsd) [inline develop/dev-guide-sample-application-spring-boot.md:522]:
```suggestion
### Step 6.3 Requests with Shell script
```

File: develop/dev-guide-sample-application-spring-boot.md [hunk 3]
```diff
@@ -567,28 +511,28 @@ curl --location --request PUT 'http://localhost:8080/player/trade' \
...
-6. the player with the `id` of 1 is the seller and the player with the id of 2 is the buyer, buy 10 goods and cost 100 gold coins
+1. Create 10 players in a loop.
+2. Get the information of players with the `id` of 1.
+3. Get a list of up to 3 players.
+4. Get a page of players with the `index` of 0 and the `size` of 2.
+5. Get the total number of players.
+6. Perform a transaction, where the player with the `id` of 1 is the seller and the player with the `id` of 2 is the buyer, and 10 `goods` are purchased at the cost of 100 `coins`.
 
-You can run this script with `make request` or `./request.sh` command and the result should look like this:
+You can run this script with `make request` or `./request.sh` command. The result should look like this:
```
Comment (hfxsd) [inline develop/dev-guide-sample-application-spring-boot.md:533]:
```suggestion
You can run this script with `make request` or `./request.sh`. The result should look like this:
```

File: develop/dev-guide-sample-application-spring-boot.md [hunk 4]
```diff
@@ -767,19 +711,19 @@ spring:
...
 
-- `spring.datasource.url` : URL of the database connection
-- `spring.datasource.url` : database username
-- `spring.datasource.password` : the database password, this is empty, need to comment or delete
-- `spring.datasource.driver-class-name` : database driver, since TiDB is compatible with MySQL, use mysql-connector-java driver class here `com.mysql.cj.jdbc`.
-- `jpa.show-sql` : when true, the SQL run by JPA will be printed
-- `jpa.database-platform` : the selected database dialect, here we connect to TiDB, so naturally we choose **TiDB dialect**, this dialect is only available in Hibernate version `6.0.0.Beta2` and above, please notice the dependency version.
-- `jpa.hibernate.ddl-auto` : The `create-drop` selected here will create the table at the beginning of the program and delete it on exit. Please do not use this in a formal environment, but we are using this as a sample application and want to minimize the impact on the database data, so we have chosen this option.
+- `spring.datasource.url` : URL of the database connection.
+- `spring.datasource.url` : the database username.
```
Comment (hfxsd) [inline develop/dev-guide-sample-application-spring-boot.md:717]: 参数错了。猜测应该是 spring.datasource.username 吧？

File: develop/dev-guide-sample-application-spring-boot.md [hunk 5]
```diff
@@ -1093,19 +1040,19 @@ public class PlayerServiceImpl implements PlayerService {
...
+The `PlayerServiceImpl` implementation class also has a [`@Transactional`](https://docs.spring.io/spring-framework/docs/current/reference/html/data-access.html#transaction-declarative-annotations) annotation in addition to the `@Service` annotation. When transaction management is enabled in the application (which can be turned on using [`@EnableTransactionManagement`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/annotation/EnableTransactionManagement.html), but is turned on by default by `Spring Boot` and does not need to be configured manually again), `Spring` automatically wraps all objects with the `@Transactional` annotation in a proxy and uses this proxy for object invocation processing.
```
Comment (hfxsd) [inline develop/dev-guide-sample-application-spring-boot.md:1045]:
```suggestion
The `PlayerServiceImpl` implementation class also has a [`@Transactional`](https://docs.spring.io/spring-framework/docs/current/reference/html/data-access.html#transaction-declarative-annotations) annotation in addition to the `@Service` annotation. When transaction management is enabled in the application (which can be turned on using [`@EnableTransactionManagement`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/annotation/EnableTransactionManagement.html), but is turned on by default by `Spring Boot`. You don not need to manually configure it.), `Spring` automatically wraps all objects with the `@Transactional` annotation in a proxy and uses this proxy for object invocation processing.
```

File: develop/dev-guide-sample-application-spring-boot.md (hunk 4 above)
Comment (ran-huang) [reply develop/dev-guide-sample-application-spring-boot.md:717]: 是的，nice catch

Comment (ran-huang) [inline develop/dev-guide-sample-application-spring-boot.md:717]:
```suggestion
- `spring.datasource.username` : the database username.
```

**Review by hfxsd — APPROVED** [review]

**Pull request merged**